packages/mds-db/events.ts

```diff
@@ -25,7 +25,7 @@ import type { ReadEventsQueryParams, ReadEventsResult } from './types'
 
 export async function writeEvent(event: VehicleEvent) {
   const client = await getWriteableClient()
-  const device = IngestServiceClient.getDevice({ device_id: event.device_id, provider_id: event.provider_id })
+  const device = await IngestServiceClient.getDevice({ device_id: event.device_id, provider_id: event.provider_id })
```

Comment:
…but, `device` promises to be defined. I wonder if `isDefined` could be typed? `isDefined<T>(value: T | null | undefined): boolean` So, you could specifically test `isDefined<DeviceDomainModel>(device)`